wled00/FX.cpp

```diff
@@ -6497,7 +6445,10 @@ typedef struct Gravity {
 ///////////////////////
 //   * GRAVCENTER    //
 ///////////////////////
-uint16_t mode_gravcenter(void) {                // Gravcenter. By Andrew Tuline.
+// Gravcenter effects By Andrew Tuline.
+// Gravcenter base function for Gravcenter (0), Gravcentric (1), Gravimeter (2), Gravfreq (3) (merged by @dedehai)
+
+uint16_t mode_gravcenter_base(unsigned mode) {               
```

Comment:
well, merging them like this saves 1.2k of flash so actually it is quite some shared code. 
while I agree this could be made prettier I don't think seperating it in a different way would improve the readability.
the runtime checks are negligible, that is only done once per frame.